### PR TITLE
feat: add background color to have sufficient contrast on controls

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

File: packages/mux-player/src/template.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ export const template = (props: MuxTemplateProps) => html`
           />`
         : ""}
     </mux-video>
+    <div slot="gestures-chrome" class="mxp-background"></div>
```
Comment (gkatsev) [inline packages/mux-player/src/template.ts:75]: Putting it in the `gestures-chrome` allows it to be in front of the media player but before the controls.

File: packages/mux-player/src/styles.css [hunk 2]
```diff
@@ -30,6 +30,13 @@ media-control-bar [role="switch"] {
   background-color: var(--media-control-background, rgba(20, 20, 30, 0.7));
 }
 
+.mxp-background {
+  height: 100%;
+  width: 100%;
+  position: absolute;
+  background-color: rgba(0, 0, 0, 0.6);
```
Comment (gkatsev) [inline packages/mux-player/src/styles.css:37]: design calls for 0.5, but I realized that we have text in the control bar, the time displays, and at 0.5 our contrast ratio is around 3, where-as with 0.6 we're at 5.74 which is definitely above the 4.5 WCAG requirement for text https://www.w3.org/WAI/WCAG21/Techniques/general/G18.html and the 3 req for icons https://www.w3.org/WAI/WCAG21/Techniques/general/G207.html

File: packages/mux-player/src/template.ts (hunk 1 above)
Comment (gkatsev) [reply packages/mux-player/src/template.ts]: went with a css-only solution.

File: packages/mux-player/src/styles.css [hunk 3]
```diff
@@ -30,6 +30,10 @@ media-control-bar [role="switch"] {
   background-color: var(--media-control-background, rgba(20, 20, 30, 0.7));
 }
 
+media-controller:is([media-paused], :not([user-inactive]))::part(vertical-layer) {
```
Comment (gkatsev) [inline packages/mux-player/src/styles.css:33]: target the vertical-layer part, where the controls are, when either the media is paused (and controls are showing) or when we're playing and the user is active and thus the controls are showing.

Comment (cjpillsbury) [reply packages/mux-player/src/styles.css:33]: I think we need `centered-layer` too. See: https://github.com/muxinc/media-chrome/blob/main/src/js/media-container.js#L118-L120

Comment (cjpillsbury) [reply packages/mux-player/src/styles.css:33]: nevermind, this should be fine! (we don't have a convention on which to use just yet)

**Review by cjpillsbury — APPROVED** [review]
small tweak needed I believe

File: packages/mux-player/src/styles.css (hunk 3 above)
Comment (gkatsev) [reply packages/mux-player/src/styles.css:33]: We want this to be as far behind as possible and for now, which should be the vertical layer. Plus, we're not letting mux-player customize controls yet, so, we should end up with both.

**Comment by luwes** [comment]
is there any way this can be animated with the same transition the controls use for hiding?
right now it's an abrupt show / hide, not so smooth

**Comment by gkatsev** [comment]
Good call, let me take a look at what it takes to transition it.

**Comment by gkatsev** [comment]
@luwes updated to transition in and out. The durations were taken from media-chrome's transitions for the controls. Longer term, would probably be nice to make them be css variables or something so that it could be re-used and changed in one place.

File: packages/mux-player/src/styles.css [hunk 4]
```diff
@@ -30,6 +30,15 @@ media-control-bar [role="switch"] {
   background-color: var(--media-control-background, rgba(20, 20, 30, 0.7));
 }
 
+media-controller::part(vertical-layer) {
+  transition: background-color 0.25s;
+}
+
+media-controller:is([media-paused], :not([user-inactive]))::part(vertical-layer) {
+  background-color: rgba(0, 0, 0, 0.6);
+  transition: background-color 1s;
```
Comment (cjpillsbury) [inline packages/mux-player/src/styles.css:39]: 😎  same transition rate as https://github.com/muxinc/media-chrome/blob/main/src/js/media-container.js#L98

**Review by cjpillsbury — APPROVED** [review]
re-approved

**Pull request merged**